[#65026] Project forms improvements

myabc opened the pull request:
⚠️ ~~**This PR is based on #18715, Please review/merge that PR first.**~~

# Ticket

https://community.openproject.org/wp/65026

# What are you trying to accomplish?

Visual and text improvements to bring the work closer to the Figma mockups.

## Screenshots

| Create | Create from Template | Copy |
|--------|--------|--------|
| ![Screenshot 2025-07-02 at 00 13 47](https://github.com/user-attachments/assets/ed3ddb9c-4db4-4d57-96b2-50aee573a584) | ![Screenshot 2025-07-02 at 00 13 59](https://github.com/user-attachments/assets/6ca9bad2-2fa0-46c1-b7d3-613007b60383) | ![image](https://github.com/user-attachments/assets/f912ec17-5436-43a4-a490-cc55edd61025) | 

# What approach did you choose and why?

This follows the simplest approach - maintaining a separate box with border for the copy from template/project options.

# Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

myabc committed:
Relocate copy options to top of create/copy pages

Also removes highlighting from uses template selector section.

myabc committed:
Update copy options legend wording

myabc committed:
Remove unused highlight-when-value-selected controller

HDinger reviewed (changes requested):
The code changes look fine 👍 
I just have one thing: In the AC it says: 

> On the project copy page, move the copy from project options to the top of the page (per Figma mockups). A project selector should be displayed similar to the use template selector on the create page, but with the label Project copied and read-only


That disabled project selector field is missing.

<img width="544" alt="Bildschirmfoto 2025-07-02 um 08 14 23" src="https://github.com/user-attachments/assets/28356b79-d34a-47aa-9504-1c675e01ad5d" />

myabc committed:
Implement disabled project select on copy page

Creats a `Projects::SourceProjectSelectComponent`.

myabc committed:
Inline Projects::TemplateAutocompleter form

Makes `TemplateSelectComponent` and `SourceProjectSelectComponent`
implementations consistent. `TemplateAutocompleter` is unlikely to be
reused by other components at present.

myabc committed:
Move I18n strings from 'js-en.yml'

myabc committed:
Prune more unused I18n strings from 'js-en.yml'

myabc commented:
@HDinger I've now implemented the project selector on the copy page, except for the styling (which I couldn't figure out - it's not automatically styled as disabled by setting the `disabled` input). It does behave as a disabled autocompleter however:

https://github.com/user-attachments/assets/b6ed6445-8cc4-4d9f-a28f-59d6b6a3f010

myabc commented on config/locales/en.yml:
```diff
@@ -2043,6 +2043,16 @@ en:
       failed: 'Could not copy project "%{source_project_name}" to project "%{target_project_name}".'
       succeeded: 'Copied project "%{source_project_name}" to "%{target_project_name}".'
+    source_project_label: "Project copied"
+    source_project_placeholder: "(none)"
```
This is a pretty generic translation. It looks like we can use `label_none_parentheses` instead.
